setuptools/tests/test_easy_install.py

```diff
@@ -121,6 +121,21 @@ def test_write_exception(self):
         with pytest.raises(distutils.errors.DistutilsError):
             cmd.cant_write_to_target()
 
+    def test__tmpdir(self):
+        """
+        Test that a temporary directory gets created and then destroyed.
+        """
+        dist = Distribution()
+        cmd = ei.easy_install(dist)
+        tmp_dir = None
+        with cmd._tmpdir() as tmp_dir_name:
```

Comment:
We don't usually test the private interface of these functions directly, it's better to try to hit the relevant code indirectly. `_tmpdir` seems to have pretty good code coverage already as it's used every time `easy_install` is called.

Reply:
So no tests required?

Reply:
Very good instincts to add a test, though.

I suspect there is *some* way to test this particular functionality by trying to assert that none of the files written into this directory persist when `easy_install` is done being used, but frankly we're trying to get rid of `easy_install` anyway, so it's probably not worth spending much time working out clever tests for it.